test/directives/vector_vectorlength_not_supported.f90

```diff
@@ -0,0 +1,9 @@
+! RUN: %flang -O0 -c %s 2>&1 | FileCheck %s --check-prefix=CHECK-WRONG-CLAUSE
+subroutine func1(a, b, m)
+  integer :: i, m, a(m), b(m)
+  !dir$ vector vectorlength(garbage)
```

Comment:
Let's add some more wrong forms to this test. In previous rounds of review I flagged some of these, e.g. 
 * Spaces instead of commas where multiple values are given
 * Garbage occurring after correct input 
 * Missing trailing ')'
 
We ought also to check that bad input causes flang to ignore the directive completely and not read the correct input and use it to set the metadata whilst also emitting the warning.